api/src/app/modules/public-comment/public-comment.service.ts

```diff
@@ -149,11 +209,93 @@ export class PublicCommentService extends DataService<PublicComment, Repository<
     }
 
     const recordIds = _.map(records, 'id');
-    const decryptedColumnCollection = await this.obtainDecryptedColumns(recordIds);
+    const decryptedColumnCollection = await this.obtainDecryptedColumns(
+      recordIds
+    );
     return records.map((r) => {
-      const decryptedPartial = _.find(decryptedColumnCollection, {'id': r.id});
+      const decryptedPartial = _.find(decryptedColumnCollection, { id: r.id });
       Object.assign(r, decryptedPartial);
       return this.convertEntity(r);
     });
   }
+
+  /**
+   * Returns the total number of public comments grouped by district.
+   * Used by analytics dashboard module.
+   *
+   * @param startDate - Start of date range (YYYY-MM-DD)
+   * @param endDate - End of date range (YYYY-MM-DD)
+   * @returns Promise resolving to an array of PublicCommentCountByDistrictResponse
+   */
+  async getCommentCountByDistrict(
+    startDate: string,
+    endDate: string
+  ): Promise<PublicCommentCountByDistrictResponse[]> {
+    return await this.repository
+      .createQueryBuilder('c')
+      .where('c.createTimestamp >= :startDate', { startDate })
+      .andWhere('c.createTimestamp <= :endDate', { endDate })
```

Comment:
When we count the number of public comments, it requires the `createTimestamp` to be within the selected date range I think? And there shouldn't have FOM in the "INITIAL" state that has comment I believe?

Reply:
Correct, FOM's in initial state shouldn't have comments, but you don't need to worry about that, just count the comments for the FOM within the date range.